web/js/palettes/model.js

```diff
@@ -479,12 +480,13 @@ export function palettesModel(models, config) {
   };
 
   var updateLookup = function (layerId) {
+    var legend, oldLegend, entries;
```

Comment:
You're moving these variables outside of their original scope. Why?

Reply:
Otherwise these values are undefined and when you use these features in WV, the console throws a `ReferenceError: oldLegend is not defined` error.

Reply:
There are no lines referencing these values outside of their original scope. You should add the `var` keyword inside the `loEach` loop, and you can just add it to the lines where those variables are given values. No need to make it a separate line.